spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/support/SubscriberFactory.java

```diff
@@ -53,7 +53,8 @@
 	 * Create a {@link PullRequest} for synchronously pulling a number of messages from
 	 * a Google Cloud Pub/Sub subscription.
 	 * @param subscriptionName the name of the subscription
-	 * @param maxMessages the maximum number of pulled messages
+	 * @param maxMessages the maximum number of pulled messages,
+	 * which can be null or must be a positive number
```

Comment:
I'd skip the "can be null". It's a required field in the proto, so the request will fail. You are right that our validation exempting `null` is a bug.

Reply:
Ok, I removed it. Bug was created under #2248.